### DISCUSSION TITLE
Genereal Feedback

### DISCUSSION THREAD

**Discussion by nl8-gh** [alagrede/znote-app#82] [General]
Hey there, still loving zNote and kinda even start using it more and more again. 
However, there are a few things that I recognize every time I am using it. Please consider the following points as what they are, my individual observations rather general KO criteria - however, there might be others thinking likewise but are shy to speak out or too lazy to type.

1. zNote is Powerful and the abillity to run/execute just everything from within zNote is kinda cool but scary too. Any chance to lock it down a little bit/enable things as I need them rather "Always On with open gates to the max" as soon I start it?

    e.g. 
    - use own folder for node_modules & isolate zNote modules from a users common dev environment - by default on by choice)

    
2. UI / Settings / 'Workflow'
   - There is a 'help' menu in the title bar, there is a '?' icon above the notes & there is a '?' icon in the lower left
   - There are settings in above the notes, a settings icon in the lower left, a switch for light/dark mode (also a setting, no?) 
   - A toast to let people know about keyboard shortcuts is good. But then please respect my workspace and get it out of my eyes rather being a 'perma banner' on the footer. It shoulb be in 'help' menu in the title bar (as in every app I ever used)
   - Move the (toggle) Icon for the SidePar Panel show/hide into the sidebar so it behaves like a hamburger menu. Right now its jumping. Thats not a good UI pattern as it appears "unfinished"
   - Folder Icon should be related to the Sidebar Panel as its has 0 functionality on the notes screen itself.
   - Everything related to the Note is, again, totally cluttered all over the place ('New' is left, but to lock, edit or share its max right while App Related Settings for display panels is in between. From a UI its really rocky (even though everything is beautiful its just not 'there' yet 
   
      --> Application related stuff should move into the corresponding menus within the title (you even got a help menu there but, other than 99% of application not 'about' (with link to site) no 'help' (since its sitting elsewhere). Clean it up and make things intuitive
   
   
      --> I believe you recognize System Preferences = set the light/dark mode based on a choice the user already made. IF he/she wants to change it, well, I'd add a 'settings' menu
  
3. Packages & Co
   - Every pacakge is always global or aligned to a folder or to a note or to ...? I honestly don't know.
   - Also, get rid of the icon in the menu above the "notes" - since there should be things related to THAT note I am on (at least I'd expect there not to be settings that got global impact). 
     
     I would prefer being able to habe a choice to import packages (or even different versions of the same) for either a specific note, folder or EcoSystem wide (All zNote) or Global (but then again, it should respect 1. from above (actually I dont want zNote replace my terminal - or have the same super powers without even asking since its basically 'root'.

4. Folders\Notes
   - To be honest, I do not like the way notes/folders are organized in any way. Its cumbersome, but I might be the only one. I got no other app with a UI that organzies things for me that complicated = always I start znote I have to remember that I artifically set folders on the left, but can not move notes into them, since I need to assign the folder to the note from within the note. Get the point? :)

5. Bug / feature or stupid user?
   - I can't get the ray.so example to work in any way
   - I found no way to display code blocks / lines as just what it is - as code blocks or, automagically as screenshot. Its not I want to hide it but I when writing technical docs for export, I want to preview a code block/code line.
   - can I somehow load modules from github directly (e.g. `npm i github:user/module`) 
   - If I use YAML to format a docs header, the note of the name is not displayed. It took me ages to find out I got to insert a H1 first, wait for the note to be renamed accordingly and then remove the H1 / shift it down to add my YAML. Would be a lot more intuitive if a) the note would read 
   -  e.g. 'FileName' tag if YAML is used or, as in every other app I ever used, could be renamed (context menu or simply clicking the name and edit it).
   
   
That became a book now and I am sorry - again, no bad feelings just sharing my "emotions" when using zNote. It is a massive timesaver for me, its just that it feels rocky (for me)

**Comment by alagrede** [comment]
HI @nl8-gh,
First of all, thank you for taking the time to write me your opinion 🙏 . I really appreciate it and it's thanks to people like you that I can continue to improve it.
I will do my best to take your feedback into account in the next versions and thus improve the UX.
If you want to add something to what I understood, feel free to complete/correct me.

**Below my comments**:

1. zNote is Powerful and the abillity to run/execute just everything from within zNote is kinda cool but scary too. Any chance to lock it down a little bit/enable things as I need them rather "Always On with open gates to the max" as soon I start it?

    e.g. 
    - use own folder for node_modules & isolate zNote modules from a users common dev environment - by default on by choice)

> Znote is installed on the User folder, so all npm packages installed are only for a user. (same for notes and settings)
**Windows: `C:\Users\USERNAME\AppData\Roaming\znote`**
**Mac: `/Users/USERNAME/Library/Application Support/znote`**
**Proposal: Concerning this point, I could add an option to "Trust note" and authorize to run code using bash or node API (and NPM packages installed)**
You can also change to an external NPM folder:
<img alt="npm-packages" width="550px" src="https://github.com/alagrede/znote-app/assets/5312754/6f0d389a-71bd-4c5d-97c1-a2b0486af474" />


    
2. UI / Settings / 'Workflow'
   - There is a 'help' menu in the title bar, there is a '?' icon above the notes & there is a '?' icon in the lower left
   - There are settings in above the notes, a settings icon in the lower left, a switch for light/dark mode (also a setting, no?)  
   - A toast to let people know about keyboard shortcuts is good. But then please respect my workspace and get it out of my eyes rather being a 'perma banner' on the footer. It shoulb be in 'help' menu in the title bar (as in every app I ever used)
   - Move the (toggle) Icon for the SidePar Panel show/hide into the sidebar so it behaves like a hamburger menu. Right now its jumping. Thats not a good UI pattern as it appears "unfinished"
   - Folder Icon should be related to the Sidebar Panel as its has 0 functionality on the notes screen itself.
   - Everything related to the Note is, again, totally cluttered all over the place ('New' is left, but to lock, edit or share its max right while App Related Settings for display panels is in between. From a UI its really rocky (even though everything is beautiful its just not 'there' yet 

      --> Application related stuff should move into the corresponding menus within the title (you even got a help menu there but, other than 99% of application not 'about' (with link to site) no 'help' (since its sitting elsewhere). Clean it up and make things intuitive
      
      --> I believe you recognize System Preferences = set the light/dark mode based on a choice the user already made. IF he/she wants to change it, well, I'd add a 'settings' menu


> Ok I get the point. I'm going to try to standardize and centralize the buttons according to their uses, even if I didn't get there by chance. 😅

  
3. Packages & Co
   - Every pacakge is always global or aligned to a folder or to a note or to ...? I honestly don't know.
   - Also, get rid of the icon in the menu above the "notes" - since there should be things related to THAT note I am on (at least I'd expect there not to be settings that got global impact). 
> Yes all packages are global. I understand that it is confusing given the position of the button 👍
     
     I would prefer being able to habe a choice to import packages (or even different versions of the same) for either a specific note, folder or EcoSystem wide (All zNote) or Global (but then again, it should respect 1. from above (actually I dont want zNote replace my terminal - or have the same super powers without even asking since its basically 'root'.
> Do you really need to load a specific package for a Note? With some work I can make it but, I'm not sure to use it. Maybe a "should have" feature?


4. Folders\Notes
   - To be honest, I do not like the way notes/folders are organized in any way. Its cumbersome, but I might be the only one. I got no other app with a UI that organzies things for me that complicated = always I start znote I have to remember that I artifically set folders on the left, but can not move notes into them, since I need to assign the folder to the note from within the note. Get the point? :)
> Yes, folders and tags are optional and can be used to filter your notes based on your usage. This is the same mechanism as Apple Notes for example or all standard note taking applications. But I'm agree, I can probably simplify the use of this a bit more.

5. Bug / feature or stupid user?
   - I can't get the ray.so example to work in any way
> Saw, Ray.so has changed its API. I will update its API. Thanks

   - I found no way to display code blocks / lines as just what it is - as code blocks or, automagically as screenshot. Its not I want to hide it but I when writing technical docs for export, I want to preview a code block/code line.
> What's wrong here? Can't get the export?
<img alt="export1" width="350px" src="https://github.com/alagrede/znote-app/assets/5312754/1b85b90b-ef6c-4fc0-974c-a720478cd58c" />
<img alt="export2" width="350px" src="https://github.com/alagrede/znote-app/assets/5312754/f71e33ab-56e1-40c0-9bd5-ad1c88f139b2" />

   - can I somehow load modules from github directly (e.g. `npm i github:user/module`) 
> I only call the `npm` command on your machine. If your `npm` can do this, it will probably work. I will try.

   - If I use YAML to format a docs header, the note of the name is not displayed. It took me ages to find out I got to insert a H1 first, wait for the note to be renamed accordingly and then remove the H1 / shift it down to add my YAML. Would be a lot more intuitive if a) the note would read 
   -  e.g. 'FileName' tag if YAML is used or, as in every other app I ever used, could be renamed (context menu or simply clicking the name and edit it).
> Using the first line for the title is a recent change. YAML header causes regression. I'll fix that 👍


Thanks again for your time. I hope I understood your points. Please let me know if I missed something.
I will start to integrate possible changes in the next versions. 😊

Anthony

**Reply by nl8-gh** [reply]
Thanks for taking it like a man :)

On 3) Thats by far even a should have - its a nice to have (fully understand there might be only avery few to tinker with different versions

On 5) I'd loev two options: a) hide code as of now b) hide code 'play' buttons but display the screenshot or a a static code block as in markdown - so I know fro looking at it whether its documentary purpose (show nicely formatted code as it would appear in exporting to htl or markdown or internal purpose = hide code as of now). Magic comments could be //Hide or //DisplayOnly kind of thing.

Hope that makes more sense.

**Reply by alagrede** [reply]
Hi,
Sorry need a last clarification for 5) 😅
Below an example of export. The first part show code and result. The second only show result (thanks to //hide magic comment).
What is missing ?
![export2](https://github.com/alagrede/znote-app/assets/5312754/eeb20292-d23a-430b-ae1d-96f4d67ee25a)

**Comment by nl8-gh** [comment]
It might be me using the wrong terms, let me explain it this way: 
I'd love a magic command to display static code as it appears in export but in the preview. 

But hey, thats likely not worth the efforts and also more a like to have rather than really required.
Just thinking of it in a way off "what would I want it to do for me so I can ditch e.g. Quiver and slim down my tool stack).

**Reply by alagrede** [reply]
Oh! On the preview. Understood! Everything seems clear now.
Thanks again. 🙏
I will let you know when I introduce these changes.

**Comment by alagrede** [comment]
Hi @nl8-gh,
I just pushed a new version of Znote including some of your observations. I hope this update helps you find Znote's user interface less "rocky". Let me know your feedback if I need to make any further changes.

Best regards,

**Reply by nl8-gh** [reply]
Sory, drown in stuff this week. Will check on the weekend. But already now: thanks for taking action so fast!

**Reply by alagrede** [reply]
Hi @nl8-gh,
No problem. You can see all changes on the download page while downloading.
